archfx_cloud/scripts/query_factory_map.py

```diff
@@ -0,0 +1,252 @@
+        for nn, line_slug_id in enumerate(self._line.keys()):
+            print(f"{'%' * 60}     {line_slug_id}        {'%' * 60}")
+            print(f"nn: {nn} out of {num_lines}, line_slug_id", line_slug_id)
+            result_dict: Dict = self.api.device.get(parent=line_slug_id)
+            for n, result in enumerate(result_dict["results"]):
+                name_label = result.get("label")
+                device_slug_id = result.get("slug")
+                line_slug_id = result.get("parent")
+                connector_type = result.get("connector_type")
+                state = result.get("state")
+
+                line_info: Dict = self._line[line_slug_id]
+                basic_device_info = {
+                    "name_label": name_label,
+                    "device_slug_id": device_slug_id,
+                    "line_slug_id": line_slug_id,
+                    "connector_type": connector_type,
+                    "state": state,
+                }
+                basic_device_info.update(line_info)
+                self._devices[device_slug_id] = basic_device_info
+                area_slug_id = basic_device_info.get("area_slug_id")
+                site_slug_id = basic_device_info.get("site_slug_id")
+
+                org = line_info["org"]
+                if device_slug_id not in self._tree[org][site_slug_id][area_slug_id][line_slug_id]:     # noqa
+                    self._tree[org][site_slug_id][area_slug_id][line_slug_id][device_slug_id] = basic_device_info   # noqa
+                print("*" * 40)
+                print("")
+                print(f"n: {n}, ", result)
+
+    def save_result_so_far(self):
+        with open("result5B.tmp", "wb") as writer:
+            writer.write(pickle.dumps([self._tree, self._sites, self._area, self._line, self._devices]))     # noqa
+
+
+    # ----------------------------------------------------------------------------------------------------------------
+    def display_all_devices(self):
+        self.read_result_so_far()
+        self.iter_over_devices()
+
+
+    def read_result_so_far(self):
+        with open("result5B.tmp", "rb") as reader:
+            data = reader.read()
+            result = pickle.loads(data)
+            self._tree, self._sites, self._area, self._line, self._devices = result  # noqa
+
+
+    def iter_over_devices(self):
+        for n, device_slug_id in enumerate(self._devices):
+            print("-" * 60)
+            print("")
+            print(n, self._devices[device_slug_id])
+
+
+    # ----------------------------------------------------------------------------------------------------------------
+    # convert to pandas
+    def result_to_dataframe(self):
+        record_list = []
+        for x in self._devices:
+            record_list.append(self._devices[x])
+        df = pd.DataFrame(record_list)
+        print(df)
+        df.to_csv("result5C.csv")
+
+
+if __name__ == "__main__":
+    email = "mbrady@archsys.io"
```

Comment:
You shouldn't hardcode your email, but make your class inherit from `BaseMain` provided by `archfx_cloud` package and it will be required to run the script with `--user` to give your email then. Token authentication is also possible.